### DISCUSSION TITLE
Should GSD entries allow escaped UTF-8 characters, or use the UTF-8 values directly?

### DISCUSSION THREAD

**Discussion by joshbuker** [cloudsecurityalliance/gsd-tools#201] [Q&A] [answered]
I ran into this while building the ruby-advisory-db importer - currently, UTF-8 characters such as ü are escaped into `\u00fc` or their equivalent for some entries.

See GSD-2014-7818 for example:
- https://data.gsd.id/GSD-2014-7818
- https://api.gsd.id/GSD-2014-7818

Ruby automatically parses in the escaped character and outputs the UTF-8 compatible value (i.e., it turns `\u00fc` into `ü`).

IMO, this should be the intended behavior, and the parser should catch any instances and correct them. Are there scenarios where keeping the escaped values instead of the direct UTF-8 encoding would make sense?

**Comment by kurtseifried** [comment, marked as answer]
I would say we should allow people to put UTF8 directly into the data. It's 2023, as we can, and often do, express ourselves with emojis. Or to quote ChatGPT:

🗣️💬💻👉🏽👥🆗🔣🆕8️⃣➡️📊🔜👥🔜👥💬💭💭🤔👉🏽👥💻🎉🎊👉🏽😀😂🤣😍😜😎🤩🥳

which is what it spat out when I asked it to convert the above text into emojis. Maybe not the best example but you get the idea.
